planner/core/find_best_task.go

```diff
@@ -277,17 +277,17 @@ func (ds *DataSource) getIndexCandidate(path *accessPath, prop *property.Physica
 	// When the prop is empty or `all` is false, `isMatchProp` is better to be `false` because
 	// it needs not to keep order for index scan.
 	if !prop.IsEmpty() && all {
-		for i, col := range path.index.Columns {
-			if col.Name.L == prop.Items[0].Col.ColName.L {
-				candidate.isMatchProp = matchIndicesProp(path.index.Columns[i:], prop.Items)
+		for i, col := range path.idxCols {
+			if col.Equal(nil, prop.Items[0].Col) {
```

Comment:
Is it possible `col` is nil according to the comment in `IndexInfo2Cols`?

Reply:
No, it's generated from `IndexInfo2PrefixCols`